modules/virtualisation/podman/podman.nix

```diff
@@ -0,0 +1,276 @@
+
+  machineOpts = {
+    # options here are loaded into python
+    # for simplicity, please use snake_case
+    options = {
+      active = mkOption {
+        type = types.bool;
+        default = false;
+        description =
+          "This machine should be started. Only one machine can be active at a time";
+      };
+      qemu_binary = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "''${pkgs.qemu}/bin/qemu-system-x86_64";
+        description =
+          "Use this to start VM with the qemu appropriate for your arch";
+      };
+      # Options passed to podman machine
+      # See https://docs.podman.io/en/latest/markdown/podman-machine.1.html
+      cpus = mkOption {
+        type = types.int;
+        default = 1;
+        description = "How many CPUs are assigned to the VM";
+      };
+      disk_size = mkOption {
+        type = types.int;
+        default = 100;
+        description = "Size of disk in gigabytes. Can only be increased";
+      };
+      image_path = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          Image to be used when starting the VM
+                  Can be a local path or a URL to an image.
+                  Alternatives can be found at https://fedoraproject.org/en/coreos/download'';
+        example = lib.literalExample ''
+          builtins.fetchurl "https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/38.20230819.3.0/x86_64/fedora-coreos-38.20230819.3.0-qemu.x86_64.qcow2.xz"
+        '';
+      };
+      memory = mkOption {
+        type = types.int;
+        default = 2048;
+        description = "RAM in MB to be assigned to the machine";
+      };
+    };
+  };
+
+in {
+  meta.maintainers = [ pkgs.lib.maintainers.michaelCTS ];
+  options.virtualisation.podman = {
+
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        This option enables Podman, a daemonless container engine for
+        developing, managing, and running OCI Containers on your Linux System.
+
+        It is a drop-in replacement for the {command}`docker` command.
+      '';
+    };
+
+    enableDockerSocket = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Make the Podman socket available in place of the Docker socket, so
+        Docker tools can find the Podman socket.
+
+        Podman implements the Docker API.
+      '';
+    };
+
+    enableDockerAlias = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Create an alias mapping <command>docker</command> to <command>podman</command>.
+      '';
+    };
+
+    #    enableNvidia = mkOption {
+    #      type = types.bool;
+    #      default = false;
+    #      description = ''
+    #        Enable use of NVidia GPUs from within podman containers.
+    #      '';
+    #    };
```

Comment:
Requires `virtualisation.containers.containersConf.settings` which is a hassle to implement and out of scope (see comment on the options in `containers.nix`). Remove comment or keep?